### PR TITLE
Sarkars/bypass ngtf

### PR DESCRIPTION
**Pull request by sayantan-nervana**

### CODE REVIEW

File: test/python/test_gather.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        val = tf.placeholder(tf.float32, shape=(5,))
+        out = tf.gather(val, [2,1])
+
+        def run_test(sess):
+            return sess.run((out,), feed_dict={val: (10.0, 20.0, 30.0, 40.0, 50.0)})[0]
+
+        assert (self.with_ngraph(run_test) == self.without_ngraph(run_test)).all()
+
+    # Vector indices, broadcast required
+    #@pytest.mark.skip(reason="Backend specific test")
```
Comment (sayantan-nervana) [inline test/python/test_gather.py]: Skip these tests, since they do not run on all backends

File: src/ngraph_builder.cc [hunk 2]
```diff
@@ -15,9 +15,12 @@
  *******************************************************************************/
 
 #include "ngraph_builder.h"
+#include <dlfcn.h>
```
Comment (sayantan-nervana) [inline src/ngraph_builder.cc]: TODO: is this still needed?

**Comment by sayantan-nervana** [comment]
TESTNOW

File: src/ngraph_builder.cc [hunk 3]
```diff
@@ -1914,6 +1916,52 @@ static Status TranslateFusedBatchNormGradOp(
...
+  std::string backend_name;
+  TF_RETURN_IF_ERROR(ngraph_bridge::GetNodeBackend(op, &backend_name));
+
+  if (backend_name != "NNPI") {
+    return errors::Internal("In translating GatherV2 op ", op->name(),
+                            " found requested backend ", backend_name,
+                            " which is unsupported");
+  }
+
+  ng::runtime::Backend* backend = BackendManager::GetBackend(backend_name);
```
Comment (shresthamalik) [inline src/ngraph_builder.cc]: We need to release the backend somewhere. Not sure where we could do that.

File: src/ngraph_mark_for_clustering.cc [hunk 4]
```diff
@@ -266,6 +266,10 @@ Status MarkForClustering(Graph* graph,
         TF_RETURN_IF_ERROR(GetNodeAttr(n->attrs(), "is_training", result));
         return Status::OK();
       };
+      confirmation_function_map["GatherV2"] = [](Node* n, bool* result) {
```
Comment (shresthamalik) [inline src/ngraph_mark_for_clustering.cc]: If we move the getting of current backend before the mark for clustering you can use that in the confirmation function. Will that help ?

File: src/ngraph_mark_for_clustering.cc [hunk 5]
```diff
@@ -608,6 +612,34 @@ Status MarkForClustering(Graph* graph,
...
+    string backend_env = std::string(ng_backend_env_value);
+    if (backend_env.empty() ||
+        !BackendManager::IsSupportedBackend(backend_env)) {
+      return errors::Internal("NGRAPH_TF_BACKEND: ", backend_env,
+                              " is not supported");
+    }
+    current_backend = backend_env;
+    // TODO: set backend. Then don't use current_backend
+  }
+  cout << "Found NG Backend " << current_backend << "\n";
```
Comment (shresthamalik) [inline src/ngraph_mark_for_clustering.cc]: Please remove cout

**Comment by sayantan-nervana** [comment]
TESTNOW

**Pull request merged**